comp/core/autodiscovery/scheduler/controller_test.go

```diff
@@ -70,53 +65,62 @@ func TestController(t *testing.T) {
 	// register a scheduler and see that it gets caught up
 	s1 := &scheduler{}
 	ms.Register("s1", s1, true)
-	s1.mutex.Lock()
-	require.ElementsMatch(t, []event{{true, "one"}, {true, "two"}}, s1.events)
-	s1.mutex.Unlock()
-
-	ms.ApplyChanges(integration.ConfigChanges{Schedule: []integration.Config{c1, c2}})
-	clk := clock.NewMock()
-	clk.Add(2000 * time.Millisecond)
+	time.Sleep(2 * time.Second)
```

Comment:
do we need this `sleep()` since after we have a `assert.EventuallyWithTf`

Reply:
Yes, it is for checking if there is any config double scheduled.